src/Microsoft.DotNet.MSBuildSdkResolver/SemanticVersion.cs

```diff
@@ -0,0 +1,181 @@
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DotNet.MSBuildSdkResolver
+{
+    internal sealed class SemanticVersion
+    {
+        public int Major { get; private set; }
+        public int Minor { get; private set; }
+        public int Patch { get; private set; }
+        public string Pre { get; private set; }
+        public string Build { get; private set; }
+
+        public SemanticVersion(int major, int minor, int patch) : this(major, minor, patch, string.Empty, string.Empty)
+        {
+        }
+
+        public SemanticVersion(int major, int minor, int patch, string pre) :
+            this(major, minor, patch, pre, string.Empty)
+        {
+        }
+
+        public SemanticVersion(int major, int minor, int patch, string pre, string build)
+        {
+            Major = major;
+            Minor = minor;
+            Patch = patch;
+            Pre = pre;
+            Build = build;
+        }
+
+        public static bool operator ==(SemanticVersion s1, SemanticVersion s2)
+        {
+            return Compare(s1, s2) == 0;
+        }
+
+        public static bool operator !=(SemanticVersion s1, SemanticVersion s2)
+        {
+            return !(s1 == s2);
+        }
+
+        public static bool operator <(SemanticVersion s1, SemanticVersion s2)
+        {
+            return Compare(s1, s2) < 0;
+        }
+
+        public static bool operator >(SemanticVersion s1, SemanticVersion s2)
+        {
+            return Compare(s1, s2) > 0;
+        }
+
+        public static bool operator >=(SemanticVersion s1, SemanticVersion s2)
+        {
+            return Compare(s1, s2) >= 0;
+        }
+
+        public static bool operator <=(SemanticVersion s1, SemanticVersion s2)
+        {
+            return Compare(s1, s2) <= 0;
+        }
+
+        public static SemanticVersion Parse(string semanticVersionString)
+        {
+            int majorSeparator = semanticVersionString.IndexOf(".");
+            if (majorSeparator == -1)
+            {
+                return null;
+            }
+
+            int major = 0;
+            if (!int.TryParse(semanticVersionString.Substring(0, majorSeparator), out major))
+            {
+                return null;
+            }
+
+            int minorStart = majorSeparator + 1;
+            int minorSeparator = semanticVersionString.IndexOf(".", minorStart);
+            if (minorSeparator == -1)
+            {
+                return null;
+            }
+
+            int minor = 0;
+            if (!int.TryParse(semanticVersionString.Substring(minorStart, minorSeparator - minorStart), out minor))
+            {
+                return null;
+            }
+
+            int patch = 0;
+            int patchStart = minorSeparator + 1;
+            int patchSeparator = semanticVersionString.FindFirstNotOf("0123456789", patchStart);
```

Comment:
Looks like this will parse 1.0.0beta as 1.0.0-beta instead of as an error. I think this should just be IndexOfAny passing '-' or '+'.